### ISSUE TITLE
No "Finish editing" button

### ISSUE THREAD

**Issue by MapperGaja** [Zaczero/osm-relatify#46]
I tried a route to edit, but I don't save/upload, because there is no blue **Finish editing** button.
What can I do, or what I need to upload my edits?

**Comment by Zaczero** [comment]
At the moment, the route must have no warnings or errors to be uploaded. You can view the current warnings on the menu on the left side (where the blue button will appear). If any of the warnings are unclear, please let me know so I can improve them. :slightly_smiling_face:

**Comment by MapperGaja** [comment]
Sorry, but this is all, what I see:

![Relatify](https://github.com/Zaczero/osm-relatify/assets/144342023/e5fc155e-7f9e-4622-a850-f993b780ef7f)

Is there a warning? I can't see nothing...

**Comment by Zaczero** [comment]
Oh, that's interesting. I'll take a look at this soon - hopefully today. Thank you for reporting this!

**Comment by Zaczero** [comment]
The routing algorithm struggles to calculate this route and it times out. I have an idea on how to improve it but haven't done so yet.

**Comment by MapperGaja** [comment]
OK. Thank you for your help!

**Comment by InsertUser** [comment]
Would it be possible to enable the "download osc" option without clearing all errors? It can be frustrating if you have done some editing only to realise that the relevant stops are missing from OSM.

**Comment by Zaczero** [comment]
> Would it be possible to enable the "download osc" option without clearing all errors? It can be frustrating if you have done some editing only to realise that the relevant stops are missing from OSM.

There is a reload button in the top-right part of the menu for cases like these.

**Comment by MapperGaja** [comment]
Hi! Sorry, I'm in Vacation, so I don't read my e-mails regularly...
So, I reload the relation an I have 'Finish edition' button, an I can save my modifications.
But the representation of relation is not the same as other relations, it isn't drawn with broken line:
![kép](https://github.com/Zaczero/osm-relatify/assets/144342023/ebd84219-890a-4782-87f2-f3f13319bac2)
An other relation from this island so:
![kép](https://github.com/Zaczero/osm-relatify/assets/144342023/754ee26b-f9ba-4d5d-aca5-e53c2b439ff5)
Is this a problem, or not?

**Comment by Dimitar5555** [comment]
I don't see an end point in the first screenshot.

**Comment by Zaczero** [comment]
@MapperGaja It's very interesting; I'll investigate it. :slightly_smiling_face:

**Comment by MapperGaja** [comment]
I have troubles with relation 15949485, too. In OSM it seem OK. The OSM Relation Analyzer (http://ra.osmsurround.org/index) says "Great! This relation seems ok."
Then, What's the problem???

**Comment by Zaczero** [comment]
@MapperGaja I fixed some OSM tagging and Relatify seems to work well now: https://www.openstreetmap.org/changeset/141383370.

**Comment by MapperGaja** [comment]
Thank you for your work!

**Comment by MapperGaja** [comment]
Now (26th January 2024) is the same status with relation #6845924, as I saw on 11th September 2023 (see above): there is no moving stripes on the orange lines. What could be the problem?

**Comment by Dimitar5555** [comment]
There are quite a few detours which slows down the pathfinding algorithm. Also, none of the turning loops have oneway tags. If they had, Relatify would've been able to find the proper route a bit faster.

**Comment by Zaczero** [comment]
Exactly! The missing moving stripes indicate a too complex path, and this is an algorithm limitation I am aware of. Currently, the interface lacks a clear "routing timeout" error message, which I am sorry for! It's definitely on my TODO list, but I am currently all-in on my OpenStreetMap NextGen project. If there are indeed oneway roads, marking them so would speed up calculation immensely. However, if there are not, then there is not much you can do than just edit the route using a different editor.

**Comment by MapperGaja** [comment]
OK, I understood. No problem. Thanks!
